docs/nmap.xsl

```diff
@@ -356,10 +356,18 @@
       function toggle(divID) {
         var item = document.getElementById(divID);
         if (item) {
-          item.className=(item.className=='hidden')?'unhidden':'hidden';
+          item.className = (item.className=='hidden')?'unhidden':'hidden';
         }
       }
-           
+
+      function toggleMetrics(divID) {
+        var item = document.getElementById(divID);
+        if (item) {
+          item.className = (item.className=='hidden')?'unhidden':'hidden';
+          document.getElementsByClassName("noprintMetrics")[0].innerHTML=(item.className=='hidden')?"(click to expand)":"(click to close)";
```

Comment:
This here is the problem. `document.getElementsByClassName("noprintMetrics")[0]` will get the *first* element with the class name "noprintMetrics". We want to get the *closest* element instead, or rather the appropriate element that was clicked. We can find that element in a couple of different ways:

1. It is the element that was clicked, so it is passed to the `onclick` handler as the `this` object.
2. We may be able to traverse the DOM from `item` up to the appropriate "click to expand" label.

I prefer the first option. Be sure you are testing with a scan file containing multiple hosts, and that you are clicking on the hidden section for each of them. This is how you can see the problem with the current approach. If you make it a general enough solution, we can use it anywhere that "click to expand" is used.